src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java

```diff
@@ -516,7 +525,12 @@ protected void parseCreateField(ParseContext context, int dimension) throws IOEx
             context.doc().add(point);
             addStoredFieldForVectorField(context, fieldType, name(), point.toString());
         } else if (VectorDataType.FLOAT == vectorDataType) {
-            Optional<float[]> floatsArrayOptional = getFloatsFromContext(context, dimension);
+            Optional<float[]> floatsArrayOptional;
+            if (isFaissSQfp16(fieldType().getKnnMethodContext())) {
```

Comment:
Will this work with model based types (i.e. IVF?)

Reply:
Discussed offline. This will depend on #1342 - Im working on this now. Lets add a todo statement in relevant place to add support for model-based indices and link to #1342.